README.md

```diff
@@ -1,4 +1,94 @@
+# Temporal .NET SDK
 
 ⚠️ UNDER ACTIVE DEVELOPMENT
 
-(for the previous .NET SDK repo, see https://github.com/temporalio/experiment-dotnet)
\ No newline at end of file
+(for the previous .NET SDK repo, see https://github.com/temporalio/experiment-dotnet)
+
+## Quick Start
+
+TODO
+
+## Usage
+
+TODO
+
+## Development
+
+### Build
+
+With `dotnet` installed with all needed frameworks and Rust installed (i.e.
+`cargo` on the `PATH`), run:
+
+    dotnet build
+
+Or for release:
+
+    dotnet build --configuration Release
+
+### Code formatting
+
+Install [CSharpier](https://csharpier.com):
+
+    dotnet tool install --global csharpier
+
+Run `dotnet format` then `csharpier`:
+
+    dotnet format style && dotnet format analyzers && dotnet csharpier .
+
+### Testing
+
+Run:
+
+    dotnet test
+
+Can add options like:
+
+* `--logger "console;verbosity=detailed"` to show logs
+  * TODO(cretz): This doesn't show Rust stdout. How do I do that?
+* `--filter "FullyQualifiedName=Temporalio.Tests.Client.TemporalClientTests.ConnectAsync_Connection_Succeeds"` to run a
+  specific test
+
+### Rebuilding Rust extension and interop layer
+
+To rebuild DLL, make sure `protoc` is on the `PATH`, then:
+
+    cargo build --manifest-path src/Temporalio/Bridge/Cargo.toml
+
+To regen core interop from header, install
+[ClangSharpPInvokeGenerator](https://github.com/dotnet/ClangSharp#generating-bindings) like:
+
+    dotnet tool install --global ClangSharpPInvokeGenerator
+
+Then, run:
+
+    ClangSharpPInvokeGenerator @src/Temporalio/Bridge/GenerateInterop.rsp
+
+### Regenerating protos
+
+Must have `protoc` on the `PATH`, then:
+
+    dotnet run --project src/Temporalio.Api.Generator
```

Comment:
Again... When/Who shold be doing this? Commited to git or gitignore?

Reply:
Developer should do this on API update. There will be a CI step that ensures that what's in repo matches what would be generated when re-run. We do want the generated code committed so anyone can just clone + build and so that it's easy to browse the code.